crates/bevy_ecs/src/query/state.rs

```diff
@@ -160,16 +329,15 @@ impl<D: QueryData, F: QueryFilter> QueryState<D, F> {
         access: &mut Access<ArchetypeComponentId>,
     ) -> Self {
         let mut state = Self::new_uninitialized(world);
-        for archetype in world.archetypes.iter() {
-            // SAFETY: The state was just initialized from the `world` above, and the archetypes being added
-            // come directly from the same world.
+        state.update_archetypes_with(world, |view, archetype| {
+            // SAFETY: The validate_world call ensures that the world is the same the QueryState
```

Comment:
nit: I don't see the `validate_world` call you are referring to, I suppose it happens in `Self::new_uninitialized`, but it's not clear.

Reply:
nvm I see where it is called (in `update_archetypes_with`)